README.md

```diff
@@ -218,29 +218,70 @@ Authors should respect the `prefers-reduced-motion` media query by conditionally
+
+If the predecessor page can't handle being restored, it may set the `irreversible` field in the activation options as an escape hatch to ensure it's unloaded. This prevents adoption and causes the predecessor to be ineligible for bfcache. Furthermore, eligibility for bfcache is a requirement for being adoptable.
+
+Since whether a portal element is hosting a context changes over time, the element itself can be thought of as having a lifecycle. This is exposed to the page via `state`.
+
+State | Meaning
+----- | -------
+empty | Nothing has ever been loaded in the portal, or the previous contents were discarded due to failed restoration, the user agent reclaiming resources, etc.
+live | The portal is hosting a browsing context.
+activated | The portal has activated. This host page is adopted or in bfcache.
+frozen | The portal is hosting a browsing context, but that context has been [frozen](https://github.com/WICG/page-lifecycle).
+epitaph | The previous contents were discarded, but unlike the empty state, there is some visual representation of the previous content.
+
+The events fired on a portal element to notify of its lifecycle changes are as follows.
+
+Event | Meaning
+----- | -------
+onactivation | The portal contents have been activated by something other than an explicit call to the `activate` method (e.g. default click, back navigation).
+onrestoration | The previous contents were restored into this portal element.
+ondiscard | The portal contents have been discarded due to a need to reclaim resources or due to the inability to perform restoration.
+onfreeze/onresume | Indicates when the portal context has been [frozen/resumed](https://github.com/WICG/page-lifecycle).
+
+There is one more event, `window.onportaladopt`, needed to communicate to a page that it has become a portal when other forms (`activate` promise resolving or `onactivation` of a predecessor portal) are not applicable.
+
+Depending on the semantics of the page, the activation of a predecessor may be viewed as traversing session history. For example, if scrolling a preview portal into view activates the preview, then scrolling back up could be considered a back navigation. A page can specify how a portal activation should affect session history by setting the `history` field in the activation options to one of `push` (default), `replace`, `back`, `forward`.
```

Comment:
This came up in a previous discussion, but instead of calling the default option "push," perhaps just "default" is fine? It seemed awkward to me at first, but it would be consistent with the naming in
https://html.spec.whatwg.org/multipage/browsing-the-web.html#history-handling-behavior

Reply:
We could also change the HTML spec to use "push", since the history handling behavior enum is not web-developer exposed.

I'm on the fence about which I would actually like more, for an API.